src/file/watcher.go

```diff
@@ -122,16 +122,14 @@ func (w *Watcher) watchFsEvents() {
 }
 
 func (w *Watcher) onEvent(event watcher.Event) bool {
-	if event.IsDir() {
-		if isEventType(event.Op, watcher.Create) {
-			w.fsWatcher.Add(event.Path)
-		}
-		return false
-	}
 	events := map[string]Event{}
-	for _, event := range w.translateEvent(event) {
-		events[event.Path] = event
+	for _, e := range w.translateEvent(event) {
+		events[e.Path] = e
 	}
+	if len(events) == 0 {
+		return false
+	}
+
 	drainTimer := time.NewTimer(drainTimeout)
 	defer drainTimer.Stop()
 	for {
```

Comment:
The actual issue is on line [143](https://github.com/Shopify/themekit/pull/889/files#diff-a281c8cf40c688f804f3d1a0a98412d64de22cc399d2aae1682d2861bb5cf467L139-L146) where we get a second stage of events, the directory event filtering from line 125 was not duplicated there.